solana-programs/anchor/audius-data/tests/track-social-actions.ts

```diff
@@ -186,6 +213,47 @@ describe("track-actions", function () {
     );
   });
 
+  it("Delegate reposts a track", async function () {
+    const delegate = await testCreateUserDelegate({
+      adminKeypair,
+      adminStorageKeypair: adminStorageKeypair,
+      program,
+      provider,
+      ...getURSMParams(),
+    });
+
+    const tx = await addTrackRepost({
+      program,
+      baseAuthorityAccount: delegate.baseAuthorityAccount,
+      adminStoragePublicKey: adminStorageKeypair.publicKey,
+      userStorageAccountPDA: delegate.userAccountPDA,
+      userAuthorityDelegateAccountPDA: delegate.userAuthorityDelegatePDA,
+      authorityDelegationStatusAccountPDA:
+        delegate.authorityDelegationStatusPDA,
+      userAuthorityKeypair: delegate.userAuthorityDelegateKeypair,
+      handleBytesArray: delegate.handleBytesArray,
+      bumpSeed: delegate.userBumpSeed,
+      id: randomString(10),
+    });
+    const info = await getTransaction(provider, tx);
+    const instructionCoder = program.coder.instruction as BorshInstructionCoder;
+    const decodedInstruction = instructionCoder.decode(
+      info.transaction.message.instructions[0].data,
+      "base58"
+    );
+    const userHandle = String.fromCharCode(...delegate.handleBytesArray);
+    const instructionHandle = String.fromCharCode(
+      ...decodedInstruction.data.userHandle.seed
+    );
+    assert.equal(instructionHandle, userHandle);
```

Comment:
for my own edification here - why do we assert on the handle being the same? we just want to make sure the repost was added for the right user?

Reply:
yup